src/core/src/op/range.cpp

```diff
@@ -87,258 +95,55 @@ void op::v4::Range::validate_and_infer_types() {
-}
-
-template <typename T>
-bool get_casted_value(const HostTensorPtr& tensor, T* val) {
-    switch (tensor->get_element_type()) {
-    case element::Type_t::bf16:
-        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::bf16>());
-        break;
-    case element::Type_t::f16:
-        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::f16>());
-        break;
-    case element::Type_t::f32:
-        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::f32>());
-        break;
-    case element::Type_t::i8:
-        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::i8>());
-        break;
-    case element::Type_t::i32:
-        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::i32>());
-        break;
-    case element::Type_t::i64:
-        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::i64>());
-        break;
-    case element::Type_t::u8:
-        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::u8>());
-        break;
-    case element::Type_t::u32:
-        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::u32>());
-        break;
-    case element::Type_t::u64:
-        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::u64>());
-        break;
-    default:
-        return false;
-    }
-    return true;
-}
-
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace rangeop {
-namespace {
-template <element::Type_t ET>
-bool evaluate(const HostTensorPtr& out,
-              const HostTensorPtr& start,
-              const HostTensorPtr& stop,
-              const HostTensorPtr& step,
-              int version) {
-    using T = typename element_type_traits<ET>::value_type;
-    double start_val;
-    double stop_val;
-    double step_val;
-    if (version < 4) {
-        start_val = static_cast<double>(*start->get_data_ptr<ET>());
-        stop_val = static_cast<double>(*stop->get_data_ptr<ET>());
-        step_val = static_cast<double>(*step->get_data_ptr<ET>());
-        if (!(check_value(start_val) && check_value(stop_val) && check_value(step_val) &&
-              (step_val != static_cast<T>(0)))) {
-            return false;
-        }
-    } else {
-        if (!(get_casted_value<double>(start, &start_val) && get_casted_value<double>(stop, &stop_val) &&
-              get_casted_value<double>(step, &step_val))) {
-            return false;
-        }
-    }
-
-    int64_t out_size = 0;
-
-    if (ov::element::Type(ET).is_integral_number()) {
-        start_val = std::trunc(start_val);
-        stop_val = std::trunc(stop_val);
-        step_val = std::trunc(step_val);
-    }
-
-    int64_t steps = static_cast<int64_t>(std::ceil(double(stop_val - start_val) / step_val));
-    if (steps > 0) {
-        out_size = steps;
-    }
-    ov::Shape out_shape = ov::Shape({static_cast<size_t>(out_size)});
-    out->set_shape(out_shape);
-
-    T start_val_casted = static_cast<T>(start_val);
-    T step_val_casted = static_cast<T>(step_val);
```

Comment:
Is it not required anymore? Can't find similar checks in new version

Reply:
Or it is covered by `range_shape_infer()`?

Reply:
Yes is covered by `shape_infer` and was not removed as it was used in old evaluate.
I've also check function `check_value` and use values e.g float16 to check detect `inf` and this check not worked, but what is used in shape infer worked correctly and exception has been thrown as expected.